staging/src/k8s.io/client-go/tools/clientcmd/merged_client_builder.go

```diff
@@ -156,6 +156,8 @@ func (config *DeferredLoadingClientConfig) Namespace() (string, bool, error) {
 			}
 			if context := raw.Contexts[currentContext]; context != nil && len(context.Namespace) > 0 {
 				return ns, false, nil
+			} else if context.Namespace == "" {
+				return "default", false, nil
```

Comment:
this looks like it is undoing the entire reason for the "we got a default namespace, determine whether it was explicit or implicit" block by turning an implicit namespace back into an explicit one.

Reply:
Letting an explicit namespace in a kubeconfig file take precedence over the in-cluster config namespace was done in https://github.com/kubernetes/kubernetes/pull/44570

This PR seems to stop honoring the in-cluster config namespace at all if a kubeconfig file is used, even if the kubeconfig file does not specify a namespace at all. That's seems just as like to break people who are currently working, as it is to fix people who are currently not getting the behavior they want.

If I'm reading that correctly, that's a breaking change I would not be in favor of. People who want to set the namespace can do so by setting it explicitly in their kubeconfig file, right?

Reply:
/hold